e2etest/bgptests/infra_setup.go

```diff
@@ -336,26 +343,46 @@ func externalContainersConfigs() map[string]frrcontainer.Config {
 	return res
 }
 
-func hostnetContainerConfig(image string) map[string]frrcontainer.Config {
-	res := make(map[string]frrcontainer.Config)
-	res["ibgp-single-hop"] = frrcontainer.Config{
-		Name:  "ibgp-single-hop",
-		Image: image,
-		Neighbor: frrconfig.NeighborConfig{
-			ASN:      metalLBASN,
-			Password: "ibgp-test",
-			MultiHop: false,
-		},
-		Router: frrconfig.RouterConfig{
-			ASN:      metalLBASN,
-			BGPPort:  179,
-			Password: "ibgp-test",
+func hostnetContainerConfig(image string, bgpMode HostBGPMode) map[string]frrcontainer.Config {
+	if bgpMode == IBGPMode {
+		return map[string]frrcontainer.Config{
+			"ibgp-single-hop": {
+				Name:  "ibgp-single-hop",
+				Image: image,
+				Neighbor: frrconfig.NeighborConfig{
+					ASN:      metalLBASN,
+					Password: "ibgp-test",
+					MultiHop: false,
+				},
+				Router: frrconfig.RouterConfig{
+					ASN:      metalLBASN,
+					BGPPort:  179,
+					Password: "ibgp-test",
+				},
+				Network:  "host",
+				HostIPv4: hostIPv4,
+				HostIPv6: hostIPv6,
+			},
+		}
+	}
+
+	return map[string]frrcontainer.Config{
+		"ebgp-single-hop": {
+			Name:  "ebgp-single-hop",
+			Image: image,
+			Neighbor: frrconfig.NeighborConfig{
+				ASN:      metalLBASN,
+				MultiHop: false,
+			},
+			Router: frrconfig.RouterConfig{
+				ASN:     externalASN,
+				BGPPort: 179,
+			},
+			Network:  "host",
+			HostIPv4: hostIPv4,
+			HostIPv6: hostIPv6,
```

Comment:
maybe to check that the input is valid we can use `switch` here and 
on default return an error?

because now, if a user wants "ibgp" and misspelled as "obgp" he will get "ebgp"

Reply:
done, and added validation on the translation